src/module/hotKeysModule.ts

```diff
@@ -51,25 +51,43 @@ export class HotKeysModule {
     settings: Gio.Settings;
     actionIdToNameMap: Map<any, any>;
     actionNameToActionMap: Map<any, any>;
-    connectId: number;
-    lastStash: number | null;
-    nextStash: number | null;
+    lastWorkspaceIndex: number | null;
+    nextWorkspaceIndex: number | null;
+    lastTileableIndex: number | null;
+    nextTileableIndex: number | null;
+    lastFocusedWindowIndex: number | null;
+    switchWorkspaceConnectionId: number;
+    tilableFocusChangeConnectionId: number;
 
     constructor() {
         this.workspaceManager = global.workspace_manager;
         this.settings = getSettings('bindings');
         this.actionIdToNameMap = new Map();
         this.actionNameToActionMap = new Map();
-        this.lastStash = null;
-        this.nextStash = null;
-
-        this.resetStash();
-        this.connectId = global.window_manager.connect(
+        this.resetWorkspaceIndexes();
+        this.resetTileableIndexes();
+        this.switchWorkspaceConnectionId = global.window_manager.connect(
             'switch-workspace',
             (_, from: number, _to: number) => {
-                if (this.lastStash !== null && from != this.lastStash) {
-                    this.resetStash();
+                if (
+                    this.lastWorkspaceIndex !== null &&
+                    from != this.lastWorkspaceIndex
+                ) {
+                    this.resetWorkspaceIndexes();
                 }
+
+                this.resetTileableIndexes();
+                Me.msWorkspaceManager.msWorkspaceList[from].disconnect(
+                    this.tilableFocusChangeConnectionId
+                );
+                this.connectTileableFocusChange();
+            }
+        );
+        const workspaceAdditionConnectionId = this.workspaceManager.connect(
```

Comment:
I havent't figured out any other way to determine that the extension is loaded and `Me.msWorkspaceManager.getActiveMsWorkspace()` doesn't return `undefined`. I tried using `this.workspaceManager.get_active_workspace().connect('tileable-focus-changed'(...))`, but it only made the extension crash I believe.